packages/rum-core/src/domain/assembly.ts

```diff
@@ -182,7 +184,11 @@ export function startRumAssembly(
         }
 
         const serverRumEvent = combine(rumContext as RumContext & Context, rawRumEvent) as RumEvent & Context
-        serverRumEvent.context = combine(commonContext.context, customerContext)
+        serverRumEvent.context = combine(
+          commonContext.context,
+          customerContext,
+          serverRumEvent.type !== 'view' ? viewContext.customerContext : undefined
+        )
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** `viewContext.customerContext` would override `customerContext`. I think it is better to do the opposite to be consistent with global context.‏

Reply:
This should be fixed.